lib/factory_bot/trait.rb

```diff
@@ -26,6 +29,13 @@ def ==(other)
 
     protected
 
+    def ensure_trait_not_self_referencing(evaulated_block)
+      if evaulated_block.respond_to?(:name) && evaulated_block.name.to_s == @name
```

Comment:
I've worked myself into the wild meta-ness of factory bot. I'm not sure what this should be and where this method should be.